src/test/java/org/fusesource/jansi/AnsiTest.java

```diff
@@ -64,6 +65,32 @@ public void apply(Ansi ansi) {
         }).toString());
     }
 
+    @ParameterizedTest
+    @CsvSource({
+        "-2147483648,ESC[2147483647T", "2147483647,ESC[2147483647S",
+        "-100000,ESC[100000T", "100000,ESC[100000S"
+    })
+    public void testScrollUp(int x, String expected) {
+        try {
+          assertAnsi(expected, Ansi.ansi().scrollUp(x));
+        } catch (StackOverflowError e) {
+          fail("Infinite loop detected.");
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "-2147483648,ESC[2147483647S", "2147483647,ESC[2147483647T",
+        "-100000,ESC[100000S", "100000,ESC[100000T"
+    })
+    public void testScrollDown(int x, String expected) {
+        try {
+          assertAnsi(expected, Ansi.ansi().scrollDown(x));
```

Comment:
Same here...

Reply:
I agree as the error will throw out anyway. I have changed that. Thanks for your suggestion.